module-apps/application-settings/windows/display-keypad/WallpaperWindow.cpp

```diff
@@ -32,14 +44,11 @@ namespace gui
                     return true;
                 },
                 this,
-                Switch ? gui::option::SettingRightItem::Checked : gui::option::SettingRightItem::Disabled));
-        };
-
-        addCheckOption(utils::translate("app_settings_display_wallpaper_logo"), isWallpaperLogoSwitchOn);
-        addCheckOption(utils::translate("app_settings_display_wallpaper_clock"), isWallpaperClockSwitchOn);
-        addCheckOption(utils::translate("app_settings_display_wallpaper_quotes"), isWallpaperQuotesSwitchOn);
+                wallpaperOption == option ? gui::option::SettingRightItem::Checked
+                                          : gui::option::SettingRightItem::Disabled));
+        }
 
-        if (isWallpaperQuotesSwitchOn) {
+        if (wallpaperOption == WallpaperOption::Quote) {
```

Comment:
this seems like part of model we are missing :)
generally, it might be nice to have one common model for lock windows and use its derivatives (or even itself) with two different windows.
Or even - by different settings, by model choose different views.
Saying that - I understand why it's not like that :)

Reply:
Model added